azure_functions_worker/shared_memory_data_transfer/shared_memory_constants.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+class SharedMemoryConstants:
+    # Directories in Unix where the memory maps can be found
+    UNIX_TEMP_DIRS = ["/dev/shm"]
+
+    # Suffix for the temp directories containing memory maps in Unix
+    UNIX_TEMP_DIR_SUFFIX = "AzureFunctions"
+
+    # The length of a bool which is the length of the part of the header flag
+    # specifying if the memory map is already created and used.
+    # This is to distinguish between new memory maps and ones that were
+    # previously created and may be in use already.
+    MEM_MAP_INITIALIZED_FLAG_NUM_BYTES = 1
+
+    # The length of a long which is the length of the part of the header
+    # specifying content length in the memory map.
+    CONTENT_LENGTH_NUM_BYTES = 8
+
+    # The total length of the header
+    CONTENT_HEADER_TOTAL_BYTES = MEM_MAP_INITIALIZED_FLAG_NUM_BYTES + \
+        CONTENT_LENGTH_NUM_BYTES
+
+    # A flag to indicate that the memory map has been initialized, may be in use
+    # and is not new.
+    # This represents a boolean value of True.
+    MEM_MAP_INITIALIZED_FLAG = b'\x01'
+
+    # A flag to indicate that the memory map has not yet been initialized.
+    # This represents a boolean value of False.
+    MEM_MAP_UNINITIALIZED_FLAG = b'\x00'
+
+    # Minimum size (in number of bytes) an object must be in order for it to be
+    # transferred over shared memory.
+    # If the object is smaller than this, gRPC is used.
+    # Note: This needs to be consistent among the host and workers.
+    #       e.g. in the host, it is defined in SharedMemoryConstants.cs
+    MIN_BYTES_FOR_SHARED_MEM_TRANSFER = 1024 * 1024  # 1 MB
+
+    # Maximum size (in number of bytes) an object must be in order for it to be
+    # transferred over shared memory.
+    # This limit is imposed because initializing objects like greater than 2GB
+    # is not allowed in DotNet.
+    # Ref: https://stackoverflow.com/a/3944336/3132415
+    # Note: This needs to be consistent among the host and workers.
+    #       e.g. in the host, it is defined in SharedMemoryConstants.cs
+    MAX_BYTES_FOR_SHARED_MEM_TRANSFER = 2 * 1024 * 1024 * 1024  # 2 GB
+
+    # This is what the size of a character is in DotNet. Can be verified by
```

Comment:
Mention UTF-16?

Reply:
Umm the encoding could be any as long as it fits in the two bytes and we are using utf-8 so I'll not mention anything here to avoid confusion.
Btw, utf-8 should be fine right? As done [here](https://github.com/Azure/azure-functions-python-worker/blob/62deedc04d6f82997799045e5fb038e16e88fe63/azure_functions_worker/shared_memory_data_transfer/shared_memory_manager.py#L92) and [here)(https://github.com/Azure/azure-functions-python-worker/blob/62deedc04d6f82997799045e5fb038e16e88fe63/azure_functions_worker/shared_memory_data_transfer/shared_memory_manager.py#L129).

Reply:
The issue is that 2 is not UTF-8.

Reply:
But UTF-8 is working though and we also have UTF-8 on the host side ([here](https://github.com/Azure/azure-functions-host/blob/ca35b39021006117687a7b4ff94f046731f959e4/src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryManager.cs#L241))